pkg/processor/runtime/rpc/connection/types.go

```diff
@@ -73,8 +74,37 @@ type ManagerConfigration struct {
 	SocketType                  SocketType
 	GetEventEncoderFunc         func(writer io.Writer) encoder.EventEncoder
 	Statistics                  runtime.Statistics
+
+	host     string
+	port     int
+	workerId int
+}
+
+func NewManagerConfigration(supportControlCommunication bool, waitForStart bool, socketType SocketType, getEventEncoderFunc func(writer io.Writer) encoder.EventEncoder, statistics runtime.Statistics, workerId int, mode functionconfig.TriggerWorkMode) *ManagerConfigration {
+	manager := &ManagerConfigration{
+		SupportControlCommunication: supportControlCommunication,
+		WaitForStart:                waitForStart,
+		SocketType:                  socketType,
+		GetEventEncoderFunc:         getEventEncoderFunc,
+		Statistics:                  statistics,
+		workerId:                    workerId,
+	}
+	switch mode {
+	case functionconfig.AsyncTriggerWorkMode:
+		manager.Kind = ConnectionAllocatorManagerKind
+	default:
+		manager.Kind = SocketAllocatorManagerKind
+	}
+	if manager.Kind == ConnectionAllocatorManagerKind {
+		manager.host = "127.0.0.1"
+		manager.port = portRangeBeginning + workerId
+	}
+	return manager
 }
 
 type ManagerKind string
 
 const SocketAllocatorManagerKind ManagerKind = "socketAllocator"
+const ConnectionAllocatorManagerKind ManagerKind = "connectionAllocator"
+
+const portRangeBeginning = 1337
```

Comment:
classic.
I see that we don't have a limit on `MaxWorkers`. Can it cause any issues?

Reply:
@TomerShor only if we 1337+workerId will go outside of port range/some of the ports in the range is already in use. Technically, it doesn't really make sense to have enormous amount of workers in async case, so we may add TODO to limit a worker max number for async processing